In [28]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

In [34]:
hub_df = pd.read_csv('../data/hubNashville__311__Service_Requests.csv', low_memory = False)
hub_df.head()

Request #  Status    Request Type  \
0     353977  Closed        COVID-19   
1     354524  Closed  Quick Requests   
2     354188  Closed  Quick Requests   
3     354445  Closed  Quick Requests   
4     334893  Closed   Public Safety   

                                     Subrequest Type  \
0  COVID-19 - Submit Input about The Plan to Reop...   
1                                     Quick Requests   
2                                     Quick Requests   
3                                     Quick Requests   
4                  Control Number Request for Towing   

                          Additional Subrequest Type      Date / Time Opened  \
0  COVID-19 - Submit Input about The Plan to Reop...  04/28/2020 02:33:22 PM   
1                                                NaN  04/28/2020 09:46:53 PM   
2                                                NaN  04/28/2020 05:09:02 PM   
3                                                NaN  04/28/2020 08:50:36 PM   
4                                                NaN  03/29/2020 01:08:55 AM   

       Date / Time Closed          Request Origin System of Origin ID  \
0  04/28/2020 02:33:22 PM  hubNashville Community                 NaN   
1  04/28/2020 09:46:53 PM                   Phone                 NaN   
2  04/28/2020 05:09:02 PM                   Phone                 NaN   
3  04/28/2020 08:50:36 PM                   Phone                 NaN   
4  03/29/2020 01:08:55 AM  hubNashville Community                 NaN   

  Contact Type  ...  State Issue  Parent Request  Closed When Created  \
0          NaN  ...        False             NaN                 True   
1          NaN  ...        False             NaN                 True   
2          NaN  ...        False             NaN                 True   
3          NaN  ...        False             NaN                 True   
4          NaN  ...        False             NaN                 True   

                              Address            City Council District    ZIP  \
0                                 NaN             NaN              NaN    NaN   
1                                 NaN             NaN              NaN    NaN   
2                                 NaN             NaN              NaN    NaN   
3                                 NaN             NaN              NaN    NaN   
4  Long Hollow Pike & Northcreek Blvd  GOODLETTSVILLE             10.0  37072   

   Latitude  Longitude                 Mapped Location  
0       NaN        NaN                             NaN  
1       NaN        NaN                             NaN  
2       NaN        NaN                             NaN  
3       NaN        NaN                             NaN  
4  36.32619 -86.700618  (36.32619, -86.70061799999999)  

[5 rows x 21 columns]

In [35]:
hub_df.tail()

Request #  Status                Request Type  \
348666       2699  Closed  Streets, Roads & Sidewalks   
348667       2214  Closed   Trash, Recycling & Litter   
348668        163  Closed   Trash, Recycling & Litter   
348669       3202  Closed  Streets, Roads & Sidewalks   
348670       1237  Closed  Streets, Roads & Sidewalks   

                           Subrequest Type  \
348666                            Potholes   
348667                        Cart Service   
348668  Missed Pickup - Trash Cart Service   
348669          Overgrowth on right of way   
348670          Overgrowth on right of way   

                  Additional Subrequest Type      Date / Time Opened  \
348666                               Roadway  07/26/2017 02:54:00 PM   
348667         Trash - Report a Damaged Cart  07/24/2017 04:58:00 PM   
348668  Trash - Curbside/Alley Missed Pickup  07/17/2017 07:12:00 AM   
348669                               Roadway  07/28/2017 10:07:00 AM   
348670                               Roadway  07/20/2017 07:44:00 AM   

            Date / Time Closed Request Origin System of Origin ID  \
348666  08/01/2017 08:12:00 AM          Phone                 NaN   
348667  08/09/2017 04:06:00 AM          Phone                 NaN   
348668  07/21/2017 11:58:00 AM          Phone                 NaN   
348669  08/15/2017 04:02:00 AM          Phone                 NaN   
348670  07/28/2017 04:22:00 PM          Phone                 NaN   

       Contact Type  ...  State Issue  Parent Request  Closed When Created  \
348666          NaN  ...        False             NaN                 True   
348667          NaN  ...        False             NaN                 True   
348668          NaN  ...        False             NaN                 True   
348669          NaN  ...        False             NaN                 True   
348670          NaN  ...        False             NaN                 True   

                               Address            City Council District  \
348666  BROOK HOLLOW RD & HARDING PIKE       NASHVILLE             23.0   
348667                  518 Benzing Rd         ANTIOCH             31.0   
348668             3117 PENN MEADE WAY       NASHVILLE             15.0   
348669        Cunniff Pkwy & Fonnic Dr  GOODLETTSVILLE              3.0   
348670       SHARPE AVE  & GRANADA AVE       NASHVILLE              5.0   

          ZIP  Latitude  Longitude        Mapped Location  
348666  37205  36.09441  -86.88171  (36.09441, -86.88171)  
348667  37013  36.04626  -86.69590   (36.04626, -86.6959)  
348668  37214  36.24230  -86.70428   (36.2423, -86.70428)  
348669  37072  36.28374  -86.74212  (36.28374, -86.74212)  
348670  37206  36.18901  -86.75364  (36.18901, -86.75364)  

[5 rows x 21 columns]

In [36]:
hub_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 348671 entries, 0 to 348670
Data columns (total 21 columns):
 #   Column                      Non-Null Count   Dtype  
---  ------                      --------------   -----  
 0   Request #                   348671 non-null  int64  
 1   Status                      348671 non-null  object 
 2   Request Type                348603 non-null  object 
 3   Subrequest Type             338024 non-null  object 
 4   Additional Subrequest Type  180510 non-null  object 
 5   Date / Time Opened          348671 non-null  object 
 6   Date / Time Closed          338424 non-null  object 
 7   Request Origin              348671 non-null  object 
 8   System of Origin ID         26398 non-null   object 
 9   Contact Type                26129 non-null   object 
 10  Preferred Language          0 non-null       float64
 11  State Issue                 348671 non-null  bool   
 12  Parent Request              3589 non-null    float64
 13  Closed When Cr

# Notes:

Columns with no null values: Request # (type int64), Status (object), Date / Time Opened (object), Request origin (object), State Issue (bool), Closed When Created (bool)

Preferred Language has no non-null values, can drop.

ZIP is object, should convert to int if needed.

Latitude and Longitude are floats, can convert to geo.

In [37]:
hub_df.isnull().sum()

Request #                          0
Status                             0
Request Type                      68
Subrequest Type                10647
Additional Subrequest Type    168161
Date / Time Opened                 0
Date / Time Closed             10247
Request Origin                     0
System of Origin ID           322273
Contact Type                  322542
Preferred Language            348671
State Issue                        0
Parent Request                345082
Closed When Created                0
Address                        63613
City                           67900
Council District               68097
ZIP                            67996
Latitude                       64483
Longitude                      64483
Mapped Location                64483
dtype: int64

In [38]:
hub_df['Status'].value_counts()

Closed         338421
In Progress      6510
New              2509
Assigned          894
Transferred       315
On Hold            17
Denied              2
In Process          1
Completed           1
Escalated           1
Name: Status, dtype: int64

In [39]:
hub_df['Closed When Created'].value_counts()

True     179725
False    168946
Name: Closed When Created, dtype: int64

In [40]:
hub_df['State Issue'].value_counts()

False    348665
True          6
Name: State Issue, dtype: int64

Very few issues referred to state - only 6

In [41]:
hub_df['Request Type'].value_counts()

Trash, Recycling & Litter     145146
Streets, Roads & Sidewalks     84830
Quick Requests                 64659
Property Violations            20651
Public Safety                   9733
COVID-19                        5246
Other                           4814
Electric & Water General        4354
Transit                         3692
Hands On Volunteers             2637
Permits                         1322
Parks                           1070
Planning & Zoning                250
Storm Relief                     167
Education & Libraries             22
Workforce & Jobs                   8
Public Records Request             2
Name: Request Type, dtype: int64

145,146 requests fall into trash/recycling/litter category.

Second most common category is Streets, Roads & Sidewalks with 84,830.

Convert date opened and closed columns to datetime type

In [46]:
hub_df['Date / Time Opened'] = pd.to_datetime(hub_df['Date / Time Opened'])
hub_df['Date / Time Closed'] = pd.to_datetime(hub_df['Date / Time Closed'])

In [47]:
hub_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 348671 entries, 0 to 348670
Data columns (total 21 columns):
 #   Column                      Non-Null Count   Dtype         
---  ------                      --------------   -----         
 0   Request #                   348671 non-null  int64         
 1   Status                      348671 non-null  object        
 2   Request Type                348603 non-null  object        
 3   Subrequest Type             338024 non-null  object        
 4   Additional Subrequest Type  180510 non-null  object        
 5   Date / Time Opened          348671 non-null  datetime64[ns]
 6   Date / Time Closed          338424 non-null  datetime64[ns]
 7   Request Origin              348671 non-null  object        
 8   System of Origin ID         26398 non-null   object        
 9   Contact Type                26129 non-null   object        
 10  Preferred Language          0 non-null       float64       
 11  State Issue                 348671 non-

In [48]:
hub_df.loc[hub_df['State Issue'] == True]

Request #  Status                Request Type     Subrequest Type  \
212056     148877  Closed                     Transit        State Routes   
230331     143401  Closed                     Transit        State Routes   
277059      91917  Closed                     Transit        State Routes   
346233        676  Closed                     Transit        State Routes   
348090        365  Closed  Streets, Roads & Sidewalks      Paving Request   
348540       2485  Closed  Streets, Roads & Sidewalks  Dead Animal Pickup   

       Additional Subrequest Type  Date / Time Opened  Date / Time Closed  \
212056                        NaN 2019-02-28 23:50:00 2019-03-04 19:52:00   
230331                        NaN 2019-02-18 19:17:00 2019-02-22 17:53:00   
277059                        NaN 2018-08-21 19:52:00 2018-08-22 14:22:00   
346233                        NaN 2017-07-18 15:51:00 2017-10-11 16:24:00   
348090   Request Road to be Paved 2017-07-17 12:07:00 2017-07-17 12:08:00   
348540                        NaN 2017-07-25 15:27:00 2017-07-28 08:31:00   

                Request Origin System of Origin ID Contact Type  ...  \
212056  hubNashville Community                 NaN          NaN  ...   
230331  hubNashville Community                 NaN          NaN  ...   
277059                   Phone                 NaN          NaN  ...   
346233                   Phone                 NaN          NaN  ...   
348090            Social Media                 NaN          NaN  ...   
348540                   Phone                 NaN          NaN  ...   

        State Issue  Parent Request  Closed When Created  \
212056         True             NaN                False   
230331         True             NaN                False   
277059         True             NaN                False   
346233         True             NaN                False   
348090         True             NaN                 True   
348540         True             NaN                 True   

                      Address       City Council District    ZIP  Latitude  \
212056       814 W Trinity Ln  NASHVILLE              2.0  37207  36.20681   
230331  5751 Old Hickory Blvd  HERMITAGE             14.0  37076  36.17406   
277059             Harding Pl  Nashville              NaN  37211  36.08266   
346233  4000 Old Hickory Blvd  HERMITAGE             14.0  37076  36.16746   
348090                    NaN        NaN              NaN    NaN       NaN   
348540  4300 Clarksville pike  NASHVILLE              1.0  37218  36.21958   

        Longitude        Mapped Location  
212056  -86.79697  (36.20681, -86.79697)  
230331  -86.60255  (36.17406, -86.60255)  
277059  -86.71054  (36.08266, -86.71054)  
346233  -86.60793  (36.16746, -86.60793)  
348090        NaN                    NaN  
348540  -86.83655  (36.21958, -86.83655)  

[6 rows x 21 columns]

In [50]:
hub_df.loc[hub_df['Request Type'] == 'COVID-19'].head()

Request #  Status Request Type  \
0      353977  Closed     COVID-19   
5      334780  Closed     COVID-19   
38     334789  Closed     COVID-19   
44     346122  Closed     COVID-19   
66     354331  Closed     COVID-19   

                                      Subrequest Type  \
0   COVID-19 - Submit Input about The Plan to Reop...   
5                                  COVID-19 Questions   
38                                 COVID-19 Questions   
44                                COVID-19 Violations   
66                                 COVID-19 Questions   

                           Additional Subrequest Type  Date / Time Opened  \
0   COVID-19 - Submit Input about The Plan to Reop... 2020-04-28 14:33:22   
5                                                 NaN 2020-03-28 14:17:51   
38                                                NaN 2020-03-28 14:58:01   
44                                COVID-19 Violations 2020-04-15 11:33:26   
66                                                NaN 2020-04-28 19:32:58   

    Date / Time Closed          Request Origin System of Origin ID  \
0  2020-04-28 14:33:22  hubNashville Community                 NaN   
5  2020-03-28 23:35:56  hubNashville Community                 NaN   
38 2020-04-01 19:56:36  hubNashville Community                 NaN   
44 2020-04-16 14:55:01  hubNashville Community                 NaN   
66 2020-05-08 19:13:25  hubNashville Community                 NaN   

   Contact Type  ...  State Issue  Parent Request  Closed When Created  \
0           NaN  ...        False             NaN                 True   
5           NaN  ...        False             NaN                False   
38          NaN  ...        False             NaN                False   
44          NaN  ...        False             NaN                False   
66          NaN  ...        False             NaN                False   

          Address       City Council District    ZIP  Latitude  Longitude  \
0             NaN        NaN              NaN    NaN       NaN        NaN   
5             NaN        NaN              NaN    NaN       NaN        NaN   
38            NaN        NaN              NaN    NaN       NaN        NaN   
44  928 6th Ave S  NASHVILLE             17.0  37203  36.14904 -86.772273   
66            NaN        NaN              NaN    NaN       NaN        NaN   

                     Mapped Location  
0                                NaN  
5                                NaN  
38                               NaN  
44  (36.1490396, -86.77227289999999)  
66                               NaN  

[5 rows x 21 columns]

In [51]:
hub_df['Request Origin'].value_counts()

Phone                     244656
hubNashville Community     88649
Mobile App                 13051
Email                       1721
Mayor's Office               372
Council Member               129
Council's Website             50
Walk-In                       18
Mayor's Website                8
In Person at Event             6
USPS Letter                    5
Social Media                   5
OEM Case                       1
Name: Request Origin, dtype: int64

In [63]:
# records with no zip code
nozip = hub_df[hub_df['ZIP'].isnull()]
nozip

Request #  Status                Request Type  \
0          353977  Closed                    COVID-19   
1          354524  Closed              Quick Requests   
2          354188  Closed              Quick Requests   
3          354445  Closed              Quick Requests   
5          334780  Closed                    COVID-19   
...           ...     ...                         ...   
348557        349  Closed   Trash, Recycling & Litter   
348585       2134  Closed                     Permits   
348619        438  Closed    Electric & Water General   
348622        449  Closed  Streets, Roads & Sidewalks   
348655        618  Closed   Trash, Recycling & Litter   

                                          Subrequest Type  \
0       COVID-19 - Submit Input about The Plan to Reop...   
1                                          Quick Requests   
2                                          Quick Requests   
3                                          Quick Requests   
5                                      COVID-19 Questions   
...                                                   ...   
348557                                Recycling Education   
348585                           Short Term Rental Permit   
348619                              Water Billing Concern   
348622                                 Dead Animal Pickup   
348655                                   Brush Collection   

                               Additional Subrequest Type  Date / Time Opened  \
0       COVID-19 - Submit Input about The Plan to Reop... 2020-04-28 14:33:22   
1                                                     NaN 2020-04-28 21:46:53   
2                                                     NaN 2020-04-28 17:09:02   
3                                                     NaN 2020-04-28 20:50:36   
5                                                     NaN 2020-03-28 14:17:51   
...                                                   ...                 ...   
348557                                                NaN 2017-07-17 11:41:00   
348585                                                NaN 2017-07-24 13:38:00   
348619                                                NaN 2017-07-17 14:33:00   
348622                                                NaN 2017-07-17 14:55:00   
348655                                                NaN 2017-07-18 09:22:00   

        Date / Time Closed          Request Origin System of Origin ID  \
0      2020-04-28 14:33:22  hubNashville Community                 NaN   
1      2020-04-28 21:46:53                   Phone                 NaN   
2      2020-04-28 17:09:02                   Phone                 NaN   
3      2020-04-28 20:50:36                   Phone                 NaN   
5      2020-03-28 23:35:56  hubNashville Community                 NaN   
...                    ...                     ...                 ...   
348557 2017-07-17 11:41:00                   Phone                 NaN   
348585 2017-08-04 12:56:00                   Phone                 NaN   
348619 2017-07-17 14:33:00                   Phone                 NaN   
348622 2017-07-26 10:33:00                   Phone                 NaN   
348655 2017-07-18 09:22:00                   Phone                 NaN   

       Contact Type  ...  State Issue  Parent Request  Closed When Created  \
0               NaN  ...        False             NaN                 True   
1               NaN  ...        False             NaN                 True   
2               NaN  ...        False             NaN                 True   
3               NaN  ...        False             NaN                 True   
5               NaN  ...        False             NaN                False   
...             ...  ...          ...             ...                  ...   
348557          NaN  ...        False             NaN                 True   
348585          NaN  ...        False             NaN                 True   
348619          NaN  ...  

In [69]:
nozip_closed_quick = nozip[nozip['Closed When Created'] == True]
nozip_closed_quick

Request #  Status                Request Type  \
0          353977  Closed                    COVID-19   
1          354524  Closed              Quick Requests   
2          354188  Closed              Quick Requests   
3          354445  Closed              Quick Requests   
22         334765  Closed               Public Safety   
...           ...     ...                         ...   
348557        349  Closed   Trash, Recycling & Litter   
348585       2134  Closed                     Permits   
348619        438  Closed    Electric & Water General   
348622        449  Closed  Streets, Roads & Sidewalks   
348655        618  Closed   Trash, Recycling & Litter   

                                          Subrequest Type  \
0       COVID-19 - Submit Input about The Plan to Reop...   
1                                          Quick Requests   
2                                          Quick Requests   
3                                          Quick Requests   
22                      Control Number Request for Towing   
...                                                   ...   
348557                                Recycling Education   
348585                           Short Term Rental Permit   
348619                              Water Billing Concern   
348622                                 Dead Animal Pickup   
348655                                   Brush Collection   

                               Additional Subrequest Type  Date / Time Opened  \
0       COVID-19 - Submit Input about The Plan to Reop... 2020-04-28 14:33:22   
1                                                     NaN 2020-04-28 21:46:53   
2                                                     NaN 2020-04-28 17:09:02   
3                                                     NaN 2020-04-28 20:50:36   
22                                                    NaN 2020-03-28 08:58:16   
...                                                   ...                 ...   
348557                                                NaN 2017-07-17 11:41:00   
348585                                                NaN 2017-07-24 13:38:00   
348619                                                NaN 2017-07-17 14:33:00   
348622                                                NaN 2017-07-17 14:55:00   
348655                                                NaN 2017-07-18 09:22:00   

        Date / Time Closed          Request Origin System of Origin ID  \
0      2020-04-28 14:33:22  hubNashville Community                 NaN   
1      2020-04-28 21:46:53                   Phone                 NaN   
2      2020-04-28 17:09:02                   Phone                 NaN   
3      2020-04-28 20:50:36                   Phone                 NaN   
22     2020-03-28 08:58:16  hubNashville Community                 NaN   
...                    ...                     ...                 ...   
348557 2017-07-17 11:41:00                   Phone                 NaN   
348585 2017-08-04 12:56:00                   Phone                 NaN   
348619 2017-07-17 14:33:00                   Phone                 NaN   
348622 2017-07-26 10:33:00                   Phone                 NaN   
348655 2017-07-18 09:22:00                   Phone                 NaN   

       Contact Type  ...  State Issue  Parent Request  Closed When Created  \
0               NaN  ...        False             NaN                 True   
1               NaN  ...        False             NaN                 True   
2               NaN  ...        False             NaN                 True   
3               NaN  ...        False             NaN                 True   
22              NaN  ...        False             NaN                 True   
...             ...  ...          ...             ...                  ...   
348557          NaN  ...        False             NaN                 True   
348585          NaN  ...        False             NaN                 True   
348619          NaN  ...  

In [70]:
nozip_closed_notquick = nozip[nozip['Closed When Created'] == False]
nozip_closed_notquick

Request #       Status                Request Type  \
5          334780       Closed                    COVID-19   
38         334789       Closed                    COVID-19   
52         319120  In Progress  Streets, Roads & Sidewalks   
66         354331       Closed                    COVID-19   
74         334814       Closed                    COVID-19   
...           ...          ...                         ...   
341944       6327       Closed                     Transit   
343433      20529       Closed  Streets, Roads & Sidewalks   
343799       5887       Closed                     Transit   
344477      14087       Closed                     Transit   
347112       1675       Closed  Streets, Roads & Sidewalks   

             Subrequest Type Additional Subrequest Type  Date / Time Opened  \
5         COVID-19 Questions                        NaN 2020-03-28 14:17:51   
38        COVID-19 Questions                        NaN 2020-03-28 14:58:01   
52                  Potholes                    Roadway 2020-03-06 12:07:22   
66        COVID-19 Questions                        NaN 2020-04-28 19:32:58   
74        COVID-19 Questions                        NaN 2020-03-28 16:37:16   
...                      ...                        ...                 ...   
341944          State Routes                        NaN 2017-08-14 14:35:00   
343433  Contractor Complaint                        NaN 2017-10-09 14:49:00   
343799          State Routes                        NaN 2017-08-10 20:04:00   
344477          State Routes                        NaN 2017-09-09 07:55:00   
347112  Contractor Complaint                        NaN 2017-07-21 14:17:00   

        Date / Time Closed          Request Origin System of Origin ID  \
5      2020-03-28 23:35:56  hubNashville Community                 NaN   
38     2020-04-01 19:56:36  hubNashville Community                 NaN   
52                     NaT  hubNashville Community                 NaN   
66     2020-05-08 19:13:25  hubNashville Community                 NaN   
74     2020-04-01 21:18:13  hubNashville Community                 NaN   
...                    ...                     ...                 ...   
341944 2017-10-11 16:25:00                   Phone                 NaN   
343433 2017-10-09 19:46:00                   Phone                 NaN   
343799 2017-10-11 16:25:00                   Email                 NaN   
344477 2017-10-18 15:15:00                   Email                 NaN   
347112 2018-07-09 22:05:00                   Phone                 NaN   

       Contact Type  ...  State Issue  Parent Request  Closed When Created  \
5               NaN  ...        False             NaN                False   
38              NaN  ...        False             NaN                False   
52              NaN  ...        False             NaN                False   
66              NaN  ...        False             NaN                False   
74              NaN  ...        False             NaN                False   
...             ...  ...          ...             ...                  ...   
341944          NaN  ...        False             NaN                False   
343433          NaN  ...        False             NaN                False   
343799     Resident  ...        False             NaN                False   
344477        Other  ...        False             NaN                False   
347112          NaN  ...        False             NaN                False   

                 Address City Council District  ZIP   Latitude  Longitude  \
5                    NaN  NaN              NaN  NaN        NaN        NaN   
38                   NaN  NaN              NaN  NaN        NaN        NaN   
52         516 Landau Dr  NaN              NaN  NaN  36.144152 -86.885505   
66                   NaN  NaN              NaN  NaN        NaN        NaN   
74                   NaN  NaN              NaN  NaN        NaN        NaN   
...                  ...  

In [71]:
nozip_closed_notquick['Request Type'].value_counts()

Other                         2599
COVID-19                      1723
Hands On Volunteers           1062
Streets, Roads & Sidewalks     524
Trash, Recycling & Litter      458
Transit                        381
Property Violations            274
Public Safety                  165
Electric & Water General       125
Parks                           36
Permits                         22
Planning & Zoning                9
Storm Relief                     4
Quick Requests                   4
Education & Libraries            2
Workforce & Jobs                 1
Name: Request Type, dtype: int64

In [72]:
nozip_closed_notquick[nozip_closed_notquick['Request Type'] == 'Other']

Request #      Status Request Type      Subrequest Type  \
84         334853      Closed        Other  Mayor's Office Case   
99         330588      Closed        Other  Mayor's Office Case   
158        330890    Assigned        Other  Mayor's Office Case   
159        330884    Assigned        Other  Mayor's Office Case   
195        334903      Closed        Other  Mayor's Office Case   
...           ...         ...          ...                  ...   
247479     126531      Closed        Other  Mayor's Office Case   
248062     128342      Closed        Other  Mayor's Office Case   
253466      96715    Assigned        Other                Other   
268391      91327    Assigned        Other  Mayor's Office Case   
292647      74164  In Process        Other                Other   

       Additional Subrequest Type  Date / Time Opened  Date / Time Closed  \
84                            NaN 2020-03-28 20:53:02 2020-05-01 16:59:06   
99                            NaN 2020-03-22 14:52:39 2020-04-23 18:13:33   
158                           NaN 2020-03-22 21:59:01                 NaT   
159                           NaN 2020-03-22 21:50:59                 NaT   
195                           NaN 2020-03-29 05:52:26 2020-03-31 13:25:25   
...                           ...                 ...                 ...   
247479                        NaN 2018-12-29 14:13:00 2019-11-18 19:03:00   
248062                        NaN 2019-01-05 00:41:00 2019-02-04 18:20:00   
253466                        NaN 2018-09-10 16:52:00                 NaT   
268391                        NaN 2018-08-18 14:17:00                 NaT   
292647                        NaN 2018-06-11 09:17:00                 NaT   

                Request Origin System of Origin ID Contact Type  ...  \
84      hubNashville Community                 NaN          NaN  ...   
99      hubNashville Community                 NaN          NaN  ...   
158     hubNashville Community                 NaN          NaN  ...   
159     hubNashville Community                 NaN          NaN  ...   
195     hubNashville Community                 NaN          NaN  ...   
...                        ...                 ...          ...  ...   
247479  hubNashville Community                 NaN          NaN  ...   
248062  hubNashville Community                 NaN          NaN  ...   
253466  hubNashville Community                 NaN          NaN  ...   
268391  hubNashville Community                 NaN          NaN  ...   
292647         Mayor's Website                 NaN          NaN  ...   

        State Issue  Parent Request  Closed When Created  \
84            False             NaN                False   
99            False             NaN                False   
158           False             NaN                False   
159           False             NaN                False   
195           False             NaN                False   
...             ...             ...                  ...   
247479        False             NaN                False   
248062        False             NaN                False   
253466        False             NaN                False   
268391        False             NaN                False   
292647        False             NaN                False   

                                                  Address City  \
84                                                    NaN  NaN   
99                                                    NaN  NaN   
158                                                   NaN  NaN   
159                                                   NaN  NaN   
195                                                   NaN  NaN   
...                                                   ...  ...   
247479                                                NaN  NaN   
248062                                                NaN  NaN   
253466  801 SW 2nd Ave, Gainesville, FL 32601, United ...  NaN   
268391                          

In [75]:
nozip_closed_notquick[nozip_closed_notquick['Subrequest Type'] == "Mayor's Office Case"]

Request #    Status Request Type      Subrequest Type  \
84         334853    Closed        Other  Mayor's Office Case   
99         330588    Closed        Other  Mayor's Office Case   
158        330890  Assigned        Other  Mayor's Office Case   
159        330884  Assigned        Other  Mayor's Office Case   
195        334903    Closed        Other  Mayor's Office Case   
...           ...       ...          ...                  ...   
245314     123564    Closed        Other  Mayor's Office Case   
246561     124119    Closed        Other  Mayor's Office Case   
247479     126531    Closed        Other  Mayor's Office Case   
248062     128342    Closed        Other  Mayor's Office Case   
268391      91327  Assigned        Other  Mayor's Office Case   

       Additional Subrequest Type  Date / Time Opened  Date / Time Closed  \
84                            NaN 2020-03-28 20:53:02 2020-05-01 16:59:06   
99                            NaN 2020-03-22 14:52:39 2020-04-23 18:13:33   
158                           NaN 2020-03-22 21:59:01                 NaT   
159                           NaN 2020-03-22 21:50:59                 NaT   
195                           NaN 2020-03-29 05:52:26 2020-03-31 13:25:25   
...                           ...                 ...                 ...   
245314                        NaN 2018-12-14 15:38:00 2018-12-31 21:04:00   
246561                        NaN 2018-12-17 22:24:00 2019-02-08 20:28:00   
247479                        NaN 2018-12-29 14:13:00 2019-11-18 19:03:00   
248062                        NaN 2019-01-05 00:41:00 2019-02-04 18:20:00   
268391                        NaN 2018-08-18 14:17:00                 NaT   

                Request Origin System of Origin ID Contact Type  ...  \
84      hubNashville Community                 NaN          NaN  ...   
99      hubNashville Community                 NaN          NaN  ...   
158     hubNashville Community                 NaN          NaN  ...   
159     hubNashville Community                 NaN          NaN  ...   
195     hubNashville Community                 NaN          NaN  ...   
...                        ...                 ...          ...  ...   
245314  hubNashville Community                 NaN          NaN  ...   
246561  hubNashville Community                 NaN          NaN  ...   
247479  hubNashville Community                 NaN          NaN  ...   
248062  hubNashville Community                 NaN          NaN  ...   
268391  hubNashville Community                 NaN          NaN  ...   

        State Issue  Parent Request  Closed When Created  Address City  \
84            False             NaN                False      NaN  NaN   
99            False             NaN                False      NaN  NaN   
158           False             NaN                False      NaN  NaN   
159           False             NaN                False      NaN  NaN   
195           False             NaN                False      NaN  NaN   
...             ...             ...                  ...      ...  ...   
245314        False             NaN                False      NaN  NaN   
246561        False             NaN                False      NaN  NaN   
247479        False             NaN                False      NaN  NaN   
248062        False             NaN                False      NaN  NaN   
268391        False             NaN                False      NaN  NaN   

       Council District  ZIP Latitude  Longitude  Mapped Location  
84                  NaN  NaN      NaN        NaN              NaN  
99                  NaN  NaN      NaN        NaN              NaN  
158                 NaN  NaN      NaN        NaN              NaN  
159                 NaN  NaN      NaN        NaN              NaN  
195                 NaN  NaN      NaN        NaN              NaN  
...                 ...  ...      ...        ...              ...  
245314              NaN  NaN      NaN        NaN              NaN  
2465

In [78]:
closed_notquick = hub_df[hub_df['Closed When Created'] == False]
closed_notquick

Request #       Status                Request Type  \
5          334780       Closed                    COVID-19   
6          323093       Closed  Streets, Roads & Sidewalks   
7          294849  In Progress  Streets, Roads & Sidewalks   
8          323303       Closed   Trash, Recycling & Litter   
9          295257     Assigned               Public Safety   
...           ...          ...                         ...   
348578       1978       Closed  Streets, Roads & Sidewalks   
348580        913       Closed  Streets, Roads & Sidewalks   
348593       2257       Closed  Streets, Roads & Sidewalks   
348604       3526       Closed  Streets, Roads & Sidewalks   
348647       3427       Closed   Trash, Recycling & Litter   

                             Subrequest Type  \
5                         COVID-19 Questions   
6                            Illegal Dumping   
7           Tree Maintenance on right of way   
8       Missed Pickup - Recycle Cart Service   
9                         Abandoned Vehicles   
...                                      ...   
348578                   Traffic Light Issue   
348580                  Contractor Complaint   
348593                   Traffic Light Issue   
348604                  Contractor Complaint   
348647          New Service/Activate Service   

                              Additional Subrequest Type  Date / Time Opened  \
5                                                    NaN 2020-03-28 14:17:51   
6                                                Roadway 2020-03-09 13:49:36   
7                       Tree fell in road or on sidewalk 2020-01-17 16:39:26   
8                 Recycle - Curbside/Alley Missed Pickup 2020-03-09 16:26:34   
9                                      Integration Issue 2020-01-18 21:42:38   
...                                                  ...                 ...   
348578                                               NaN 2017-07-24 13:00:00   
348580                                               NaN 2017-07-19 13:00:00   
348593                                               NaN 2017-07-25 13:00:00   
348604                                               NaN 2017-07-31 16:16:00   
348647  New Construction Start Residential Trash Service 2017-07-31 13:46:00   

        Date / Time Closed          Request Origin System of Origin ID  \
5      2020-03-28 23:35:56  hubNashville Community                 NaN   
6      2020-03-16 21:13:59                   Phone                 NaN   
7                      NaT                   Phone                 NaN   
8      2020-03-17 21:12:24                   Phone                 NaN   
9                      NaT  hubNashville Community                 NaN   
...                    ...                     ...                 ...   
348578 2017-07-24 14:55:00                   Phone         OE-00021132   
348580 2017-07-19 14:59:00                   Phone         OE-00020636   
348593 2017-07-25 19:46:00                   Phone         OE-00021185   
348604 2017-07-31 21:03:00                   Email                 NaN   
348647 2017-08-10 09:05:00                   Phone                 NaN   

       Contact Type  ...  State Issue  Parent Request  Closed When Created  \
5               NaN  ...        False             NaN                False   
6               NaN  ...        False             NaN                False   
7               NaN  ...        False             NaN                False   
8               NaN  ...        False             NaN                False   
9               NaN  ...        False             NaN                False   
...             ...  ...          ...             ...                  ...   
348578          NaN  ...        False             NaN                False   
348580          NaN  ...        False             NaN                False   
348593          NaN  ...        False             NaN                False   
348604          NaN  ...        False             NaN   

In [80]:
closed_notquick['Request Type'].value_counts()

Trash, Recycling & Litter     87473
Streets, Roads & Sidewalks    41811
Property Violations           17208
COVID-19                       4470
Other                          4154
Transit                        3579
Electric & Water General       3064
Public Safety                  2852
Hands On Volunteers            2637
Permits                         696
Parks                           596
Storm Relief                    166
Planning & Zoning               146
Education & Libraries            13
Quick Requests                    9
Workforce & Jobs                  2
Public Records Request            2
Name: Request Type, dtype: int64

In [81]:
closed_notquick[closed_notquick['Request Type'] == 'Trash, Recycling & Litter']

Request #       Status               Request Type  \
8          323303       Closed  Trash, Recycling & Litter   
10         322898       Closed  Trash, Recycling & Litter   
11         323341       Closed  Trash, Recycling & Litter   
12         323346       Closed  Trash, Recycling & Litter   
13         305782  In Progress  Trash, Recycling & Litter   
...           ...          ...                        ...   
343587       9158       Closed  Trash, Recycling & Litter   
345703       3428       Closed  Trash, Recycling & Litter   
346035       1290       Closed  Trash, Recycling & Litter   
346482       1159       Closed  Trash, Recycling & Litter   
348647       3427       Closed  Trash, Recycling & Litter   

                             Subrequest Type  \
8       Missed Pickup - Recycle Cart Service   
10                              Cart Service   
11      Missed Pickup - Recycle Cart Service   
12      Missed Pickup - Recycle Cart Service   
13                                    Litter   
...                                      ...   
343587                          Cart Service   
345703          New Service/Activate Service   
346035    Missed Pickup - Trash Cart Service   
346482                          Cart Service   
348647          New Service/Activate Service   

                               Additional Subrequest Type  Date / Time Opened  \
8                  Recycle - Curbside/Alley Missed Pickup 2020-03-09 16:26:34   
10                 Recycling - Request an Additional Cart 2020-03-08 21:06:42   
11                 Recycle - Curbside/Alley Missed Pickup 2020-03-09 16:58:32   
12                 Recycle - Curbside/Alley Missed Pickup 2020-03-09 17:02:04   
13                                                    NaN 2020-02-10 21:46:38   
...                                                   ...                 ...   
343587                      Trash - Report a Damaged Cart 2017-08-17 18:15:00   
345703  New Construction Start Residential Recycling S... 2017-07-31 13:47:00   
346035               Trash - Curbside/Alley Missed Pickup 2017-07-20 13:42:00   
346482                 Trash - Moved to Home with No Cart 2017-07-19 20:49:00   
348647   New Construction Start Residential Trash Service 2017-07-31 13:46:00   

        Date / Time Closed          Request Origin System of Origin ID  \
8      2020-03-17 21:12:24                   Phone                 NaN   
10     2020-03-17 17:13:30  hubNashville Community                 NaN   
11     2020-03-17 17:14:48                   Phone                 NaN   
12     2020-03-17 17:12:36  hubNashville Community                 NaN   
13                     NaT                   Phone                 NaN   
...                    ...                     ...                 ...   
343587 2017-08-24 09:01:00                   Phone                 NaN   
345703 2017-08-10 09:05:00                   Phone                 NaN   
346035 2017-07-26 20:58:00                   Phone                 NaN   
346482 2017-07-28 13:31:00                   Phone                 NaN   
348647 2017-08-10 09:05:00                   Phone                 NaN   

       Contact Type  ...  State Issue  Parent Request  Closed When Created  \
8               NaN  ...        False             NaN                False   
10              NaN  ...        False             NaN                False   
11              NaN  ...        False             NaN                False   
12              NaN  ...        False             NaN                False   
13              NaN  ...        False             NaN                False   
...             ...  ...          ...             ...                  ...   
343587          NaN  ...        False             NaN                False   
345703     Resident  ...        False             NaN                False   
346035     Resident  ...        False             NaN                False   
346482          NaN  ...        False             NaN   

In [82]:
closed_notquick[closed_notquick['Request Type'] == 'Streets, Roads & Sidewalks']

Request #       Status                Request Type  \
6          323093       Closed  Streets, Roads & Sidewalks   
7          294849  In Progress  Streets, Roads & Sidewalks   
14         323262       Closed  Streets, Roads & Sidewalks   
23         313474       Closed  Streets, Roads & Sidewalks   
26         301304       Closed  Streets, Roads & Sidewalks   
...           ...          ...                         ...   
348544       3674       Closed  Streets, Roads & Sidewalks   
348578       1978       Closed  Streets, Roads & Sidewalks   
348580        913       Closed  Streets, Roads & Sidewalks   
348593       2257       Closed  Streets, Roads & Sidewalks   
348604       3526       Closed  Streets, Roads & Sidewalks   

                         Subrequest Type        Additional Subrequest Type  \
6                        Illegal Dumping                           Roadway   
7       Tree Maintenance on right of way  Tree fell in road or on sidewalk   
14                       Illegal Dumping                               NaN   
23                             Sidewalks                  Damaged Sidewalk   
26                    Roadwork Complaint                               NaN   
...                                  ...                               ...   
348544               Traffic Light Issue                               NaN   
348578               Traffic Light Issue                               NaN   
348580              Contractor Complaint                               NaN   
348593               Traffic Light Issue                               NaN   
348604              Contractor Complaint                               NaN   

        Date / Time Opened  Date / Time Closed          Request Origin  \
6      2020-03-09 13:49:36 2020-03-16 21:13:59                   Phone   
7      2020-01-17 16:39:26                 NaT                   Phone   
14     2020-03-09 15:53:08 2020-03-24 17:11:57  hubNashville Community   
23     2020-02-25 21:25:30 2020-05-05 16:07:46                   Phone   
26     2020-01-31 19:54:59 2020-04-17 14:48:54  hubNashville Community   
...                    ...                 ...                     ...   
348544 2017-08-01 13:00:00 2017-08-01 14:18:00                   Phone   
348578 2017-07-24 13:00:00 2017-07-24 14:55:00                   Phone   
348580 2017-07-19 13:00:00 2017-07-19 14:59:00                   Phone   
348593 2017-07-25 13:00:00 2017-07-25 19:46:00                   Phone   
348604 2017-07-31 16:16:00 2017-07-31 21:03:00                   Email   

       System of Origin ID Contact Type  ...  State Issue  Parent Request  \
6                      NaN          NaN  ...        False             NaN   
7                      NaN          NaN  ...        False             NaN   
14                     NaN          NaN  ...        False             NaN   
23                     NaN          NaN  ...        False             NaN   
26                     NaN          NaN  ...        False             NaN   
...                    ...          ...  ...          ...             ...   
348544         OE-00022027          NaN  ...        False             NaN   
348578         OE-00021132          NaN  ...        False             NaN   
348580         OE-00020636          NaN  ...        False             NaN   
348593         OE-00021185          NaN  ...        False             NaN   
348604                 NaN          NaN  ...        False             NaN   

        Closed When Created                           Address       City  \
6                     False                 2600 Grovesnor Dr    ANTIOCH   
7                     False                  2419 Kimberly Dr  NASHVILLE   
14                    False                     4824 Foley Dr  NASHVILLE   
23                    False        Blakemore Ave & 24th Ave S  NASHVILLE   
26                    False             300 Forrest Valley Dr  NASHVILLE   
...                     ...                             